xm-commons-domain-event/src/main/java/com/icthh/xm/commons/domain/event/config/SourceConfig.java

```diff
@@ -0,0 +1,31 @@
+package com.icthh.xm.commons.domain.event.config;
+
+import com.icthh.xm.commons.domain.event.service.Transport;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+
+@Getter
+@Setter
+@Slf4j
+public class SourceConfig {
+
+    private final static String TRANSPORT_PACKAGE = Transport.class.getPackageName();
+    private boolean enabled;
+    private Class<? extends Transport> transport;
+
+    public void setTransport(String transport) throws ClassNotFoundException {
+        this.transport = getTransportClass(transport);
+    }
+
+    private Class<? extends Transport> getTransportClass(String name) throws ClassNotFoundException {
+        Class<?> transport = Class.forName(TRANSPORT_PACKAGE + ".impl." + name);
```

Comment:
how make resolving more flexible, and resolve by spring bean name,
and autowire here all beans that implement Transport